Sources/MessagingPush/MessagingPush.swift

```diff
@@ -50,7 +46,20 @@ public class MessagingPush: ModuleTopLevelObject<MessagingPushInstance>, Messagi
      is no active customer, this will fail to register the device
      */
     public func registerDeviceToken(_ deviceToken: String) {
-        implementation?.registerDeviceToken(deviceToken)
+        // Compare the new deviceToken with the one stored in globalDataStore.
+        // If they are different, proceed with registering the device token.
+        // This check helps to avoid duplicate requests, as registerDeviceToken is already called on SDK initialization.
+        if deviceToken != globalDataStore.pushDeviceToken {
+            // Call the registerDeviceToken method on the implementation.
+            // This method is responsible for registering the device token and updating the globalDataStore as well.
+            if let implementation = implementation {
+                implementation.registerDeviceToken(deviceToken)
+            } else {
+                // Update the globalDataStore with the new device token.
+                // The implementation may be nil due to lifecycle issues in wrappers SDKs.
+                globalDataStore.pushDeviceToken = deviceToken
+            }
+        }
```

Comment:
```suggestion
        // The SDK might not yet have been initialized. Save the device token so when the SDK does get initialized, the token can get registered to a profile. 
        globalDataStore.pushDeviceToken = deviceToken
        implementation?.registerDeviceToken(deviceToken)
```

Are there any consequences to simplifying the logic of this function to these 2 lines of code?

Reply:
The iOS SDK [already stores the device token to global data storage](https://github.com/customerio/customerio-ios/blob/f462325f9774dba01bbfcd1a96d2f41b461eb02e/Sources/Tracking/CustomerIOImplementation.swift#L226) every time that `registerDeviceToken` is called on the SDK. 

Therefore, after this PR gets merged with the changes made to `CustomerIO` class:
```swift
if let deviceToken = globalDAtaStore.deviceToken {
  registerDeviceToken(deviceToken)
}
```

we will be performing a redundant call even if `MessagingPush.registerDeviceToken` doesn't get called. Each time the SDK gets initialized after registering a device token the first time in the app's history, the device token will get registered again. So I am not sure the logic in this function as the PR currently has will make a difference. 

I think it's ideal to avoid redundant network calls. I would prefer it myself. I am not sure that we can accomplish that at this moment without adding a lot of extra logic to the SDK code?

Reply:
I am not sure I follow, the following line of code in `MessagingPush.registerDeviceToken` 
`if deviceToken != globalDataStore.pushDeviceToken`

It saves the extra network call. 

The later logic just saved another insertion to "GlobablDataStore" which will already happen in [the method you shared](https://github.com/customerio/customerio-ios/blob/f462325f9774dba01bbfcd1a96d2f41b461eb02e/Sources/Tracking/CustomerIOImplementation.swift#L226)

Reply:
Ok. Now that I think about it again, I can see how this would work. 

Thanks for making the comments in the function explaining why we are doing this logic.